pymc3/tests/test_distributions.py

```diff
@@ -2256,14 +2255,13 @@ def test_dirichlet_multinomial(self, n):
             dirichlet_multinomial_logpmf,
         )
 
-    @pytest.mark.xfail(reason="Distribution not refactored yet")
     def test_dirichlet_multinomial_matches_beta_binomial(self):
         a, b, n = 2, 1, 5
         ns = np.arange(n + 1)
-        ns_dm = np.vstack((ns, n - ns)).T  # covert ns=1 to ns_dm=[1, 4], for all ns...
-        bb_logp = logpt(pm.BetaBinomial.dist(n=n, alpha=a, beta=b), ns).tag.test_value
+        ns_dm = np.vstack((ns, n - ns)).T  # convert ns=1 to ns_dm=[1, 4], for all ns...
+        bb_logp = logpt(pm.BetaBinomial.dist(n=n, alpha=a, beta=b, size=2), ns).tag.test_value
         dm_logp = logpt(
-            pm.DirichletMultinomial.dist(n=n, a=[a, b], size=(1, 2)), ns_dm
+            pm.DirichletMultinomial.dist(n=n, a=[a, b], size=2), ns_dm
```

Comment:
- If I don't put a `size` of at least 2, I get a `TypeError: Cannot convert Type TensorType(int64, vector) (of Variable TensorConstant{[0 1 2 3 4 5]}) into Type TensorType(int64, scalar). You can try to manually convert TensorConstant{[0 1 2 3 4 5]} into a TensorType(int64, scalar).` Is that expected?
- When I do use a `size` > 1, I get a `AttributeError: 'ValidatingScratchpad' object has no attribute 'test_value'`, because the variable is `None`, which makes sense with this `UserWarning` that appears in the console when calling `logpt`: `/Users/alex_andorra/tptm_alex/pymc3/pymc3/aesaraf.py:357: UserWarning: No value variable found for dirichlet_multinomial_rv.out; the random variable will not be replaced.`. I don't understand why that happens though? 🤔

Reply:
I am unfamiliar with these distributions, but if `size=1` vs. `size=2` makes a difference, could it be that theres some unexpected downsizing of `(1,)` → `()` involved?

Reply:
From what I understand, `logpt` is expecting that the first and second args have the same type. What I don't understand is why the `UserWarning` appears in the first place: the second arg is `rv_values` and is informed by `ns` and `ns_dm`. 
I'm missing something and probably need @ricardoV94 or @brandonwillard knowledge of Aesara

Reply:
You don't need to worry. It's just that the logpt is tailored to be used within a Model object. The "proper" steps to call it would be:
```python
dm = pm.DirichletMultinomial.dist(n=n, a=[a, b], size=2)
dm_value = dm.type()  # This will be the variable that correspond to the values of the dm in logp
dm.tag.value_var = dm_value  # the missing step
dm_logp = logpt(dm, {dm:dm_value}).eval({dm_value: ns})
# Can also just do the following, although that's not it's main goal:
dm_logp = logpt(dm, ns)
```
I think @brandonwillard will put an option to suppress these warnings when calling logpt directly

Reply:
Oh ok, your comment seems very useful thanks! I'll try that out ASAP and keep you posted 😉 
I still have 3 tests failing 🤞

Reply:
Thanks @ricardoV94, this is updated now